ayup/portfolio/src/main/webapp/script.js

```diff
@@ -54,3 +54,62 @@ async function getRandomFactUsingAsyncAwait() {
   var a = document.getElementById('js-fact-container');
   a.parentNode.replaceChild(newFactContainer, a);
 }
+
+/** Fetches comments from the server and adds them to the DOM. */
+function loadComments() {
+  fetch('/list-comments').then(response => response.json()).then((comments) => {
+    const commentListElement = document.getElementById('comment-list');
+    console.log(comments)
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment));
+    })
+  });
+}
+
+/** Creates an element that represents a title, a description, a username and its delete button. */
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const titleElement = document.createElement('span');
+  titleElement.innerText = comment.title;
+  titleElement.className = "font-weight-bold";
+
+  const descriptionElement = document.createElement('span');
+  descriptionElement.innerText = comment.description;
+
+  const usernameElement = document.createElement('span');
+  usernameElement.innerText = "By: " + comment.username;
+
+  const deleteButtonElement = createDeleteButton()
+  deleteButtonElement.addEventListener('click', () => {
+    deleteComment(comment);
+
+    // Remove the comment from the DOM.
+    commentElement.remove();
+  });
+
+  commentElement.appendChild(titleElement);
+  commentElement.appendChild(document.createElement("br"));
+  commentElement.appendChild(descriptionElement);
+  commentElement.appendChild(document.createElement("br"));
+  commentElement.appendChild(usernameElement);
+  commentElement.appendChild(document.createElement("br"));
+  commentElement.appendChild(deleteButtonElement);
+  commentElement.appendChild(document.createElement("hr"));
+  return commentElement;
+}
+
+/** Tells the server to delete the comment. */
+function deleteComment(comment) {
```

Comment:
nit. just pass the id to this function instead of the whole `comment` object reference.

Reply:
Done.